rm: fix -vr to match GNU's output, add -R for recursive

jhscheer opened the pull request:
This
- fixes #1768.
- adds a test case for removing directory with `--verbose`
- adds an additional flag `-R` for `--recursive`
- cleans up `prompt_file()`

As discussed here https://github.com/uutils/coreutils/issues/1768#issuecomment-796162092 there was a logic bug in `handle_dir()` that prevented the correct stdout/stderr output if `--interactive` mode was not set to always.
In an attempt to fix this bug and to fix the stdout/stderr to match GNU's output, I moved the call to `remove_dir_all()` from `handle_dir()` to `remove_dir()`. Also, for simplicity, I replaced the call to `fs::remove_dir()` with `remove_dir_all()`

sylvestre commented:
I am not a big fan of having a PR (even less commits) doing several things at once.
Would it be possible to split this PR in smaller commits?
thanks

jhscheer commented:
@sylvestre Done.

jhscheer commented:
I just realised that this pull request has two bugs which are related to #1769.

1:  `rm_uu -d non_empty_dir` deletes the directory instead of stating: `cannot remove 'non_empty_dir': Directory not empty`
2: `rm -rfv non_empty_dir` does not list the deleted file(s)
```bash
$ mkdir -p non_empty_dir && touch non_empty_dir/file_a
$ rm -d non_empty_dir
removed 'non_empty_dir/file_a'
removed directory 'non_empty_dir'

$ mkdir -p non_empty_dir && touch non_empty_dir/file_a
$ rm_uu -rfv non_empty_dir
removed directory 'non_empty_dir'
```

I guess I'll split this PR into two separate PRs and fix these bugs there.
